charts/tidb-cluster/templates/tidb-initializer-job.yaml

```diff
@@ -23,22 +24,50 @@ spec:
         image: {{ .Values.mysqlClient.image }}
         imagePullPolicy: {{ .Values.mysqlClient.imagePullPolicy | default "IfNotPresent" }}
         command:
-        - /bin/sh
+        - python
         - -c
-        # Read sql from file to avoid special characters interpreted as builtin variable
-        # And also avoid plain text password show in Job and Pod spec
-        # Besides we can also add more SQL in the file for initialization, eg. create database, create user etc
         - |
-          until mysql -h {{ .Values.clusterName }}-tidb -P 4000 --connect-timeout=5 < /data/init-password.sql; do sleep 2; done
+          import os, MySQLdb
+          host = {{ printf "%s-tidb" .Values.clusterName | quote }}
+          port = 4000
+          root_password = os.environ.get('ROOT_PASSWORD')
+          system_password = os.environ.get('SYSTEM_PASSWORD')
+          conn = MySQLdb.connect(host=host, port=port, user='root', connect_timeout=5)
+          if root_password:
+              conn.cursor().execute("set password for 'root'@'%%' = %s;", (root_password,))
+          if system_password:
+              conn.cursor().execute("create user 'system'@'%%' identified by %s;", (system_password,))
```

Comment:
The system user is a special user in MySQL that cannot be created or altered. So for compatibility reasons I think we should avoid having a user named 'system'